File: src/main/java/Electro1D/Electrophoresis.java
package main.java.Electro1D;

import java.awt.Dimension;
import java.awt.FlowLayout;
import java.awt.GridBagConstraints;
import java.awt.GridBagLayout;
import java.awt.Insets;
//import java.awt.GridLayout;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.IOException;
import java.util.Vector;

import javax.swing.JButton;
import javax.swing.JFrame;
import javax.swing.JPanel;
import javax.swing.JTabbedPane;

//
import main.java.Utilities.BrowserLauncher;
import main.java.Utilities.GenomeFileParser;

/**
 * @author Bader Alharbi Desktop application The Swing version 1D
 *         Electrophoresis simulation,
 */
public class Electrophoresis extends JPanel {
	/**
	 * 
	 */
	private static final long serialVersionUID = -6625659357016475348L;
	JTabbedPane tabPane, tabPane2;
	Parameters paramPanel;
	Simulation simPanel;
	ProteinData dataPanel;
	Plot plotPanel;
	JFrame parentFrame;

	public void Header() {
		// @TODO: figure out where we want to put the help button on E1D
		JPanel headPanel = new JPanel();
		JButton help = new JButton("Help");
		help.setToolTipText("Open Help wiki for Electro1D");
		help.addActionListener(new ActionListener() {
			public void actionPerformed(ActionEvent e) {
				String url = "http://sourceforge.net/p/jbf/wiki/Electro1D";
				try {
					BrowserLauncher.openURL(url);
				} catch (IOException i) {
					System.err.println(i.getMessage());
				}
			}
		});
		headPanel.add(help);
		super.add(headPanel);
	}

	public Electrophoresis(JFrame parentFrame) {
		super.setPreferredSize(new Dimension(900, 470));
		//550(i think i want 825, 450: good size: 845, 470

    	GenomeFileParser.init();
    	
		this.parentFrame = parentFrame;
	    // to match <div id="testApplet-Electro1D-div"></div> in Electro1D_embedded.html
		this.parentFrame.setName("Electro1D");
		simPanel = new Simulation(this);
		paramPanel = new Parameters(this);
		dataPanel = new ProteinData(this);
		plotPanel = new Plot(this);
		
//        super.setSize(20,35);
		this.setLayout(new GridBagLayout());
		GridBagConstraints c = new GridBagConstraints();
		//this.setLayout(new GridLayout(1, 3, 5, 0));
		// set up left Panel
		
		JPanel tempPanel = new JPanel();
		tempPanel.setLayout(new FlowLayout(FlowLayout.LEFT));
		tempPanel.setPreferredSize(new Dimension(240, 450));
		paramPanel.setPreferredSize(new Dimension(240, 450));
		dataPanel.setPreferredSize(new Dimension(240, 450));
		tabPane = new JTabbedPane();
		tabPane.setPreferredSize(new Dimension(240, 450));
		tabPane.addTab("Parameters", paramPanel);
		tabPane.addTab("ProteinData", dataPanel);
		
		tempPanel.add(tabPane);
		
		// set up right Panel
		JPanel tempPanel2 = new JPanel();
		tempPanel2.setLayout(new FlowLayout(FlowLayout.LEFT));
		tempPanel2.setPreferredSize(new Dimension(650, 450));
		simPanel.setPreferredSize(new Dimension(650, 450));
		plotPanel.setPreferredSize(new Dimension(650, 450));		
		tabPane2 = new JTabbedPane();
		tabPane2.setPreferredSize(new Dimension(650, 450));
		tabPane2.add("Casting Tray ", simPanel);
		tabPane2.add("Plot", plotPanel);
		tempPanel2.add(tabPane2);

//        this.add(headPanel);
		c.gridx = 0;
		c.gridy = 0;
		//this.add(tempPanel, c);//tabPane
		this.add(tabPane, c);
		c.gridx = 1;
		c.gridy = 0;
		c.insets = new Insets(0, 5, 0, 0);// insets gives spacing in order of: top,left,bottom,right
		this.add(tabPane2, c);
		//this.add(tempPanel2, c);//tabPane2
		//JPanel testPanel = new JPanel();
		//this.add(testPanel);
		//testPanel.setVisible(false);
		
		paramPanel.setDefaults();
	}

		

	/**
	 * addStandard() add standards proteins
	 */
	public void addStandard() {

		simPanel.addStandard();
	}

	/**
	 * Stop running the simulation on the gel
	 */
	public void stopRun() {

		simPanel.stopRun();
	}

	/**
	 * display the simulation panel
	 */
	public void displaySim() {

		tabPane.setVisible(true);
	}

	/**
	 * set the acrylamide percentage
	 *
	 * @param acrylamide acrylamide reference
	 */
	public void setAcrylamide(Acrylamide acrylamide) {

		simPanel.setAcrylamide(acrylamide);
	}

	/**
	 * start running the simulation
	 *
	 * @param aprotein is an array of standard proteins (known)
	 * @param protein  is a unknown protein (sample)
	 * @param protein1 is a dye
	 * @param protein2 is a dye
	 */
	public void startRun(Protein aprotein[], Protein protein1, Protein protein2, Protein[] proteins) {
		simPanel.startRun(aprotein, protein1, protein2, proteins);
	}

	/**
	 * setPlotData() set the plot data
	 *
	 * @param stds is an array of standard proteins (known)
	 * @param sample  is a unknown protein (sample)
	 * @param dye is a fast-running leading marker
	 */
	public void setPlotData(Protein stds[], Protein sample, Protein dye) {
		plotPanel.setResults(stds, sample, dye);
	}

	/**
	 * displayData()display the data
	 */
	public void displayData() {
		tabPane2.setVisible(true);

	}

	/**
	 * adding the sample
	 */
	public void addSample(Protein[] stds) {

		simPanel.addSample(stds);
	}

	public void addSample2() {
		
		simPanel.addSample2();
	}
	
	public void addSampleFromFile(Vector<Protein> proteins, int wellNum) {
		simPanel.increaseDDNum();
		simPanel.addSampleFromFile(proteins, wellNum);
	}
	
	public boolean isReady() {
		return simPanel.isReady();
	}
	
	public void paramsetspeed(double d) {
		paramPanel.setSpeed(d);
	}
	
	public void updateSpeed(double d, Acrylamide acrgel) {
		simPanel.updateSpeed(d, acrgel);
	}
	
	public void setStandards(Protein[] stds) {
		simPanel.setStandards(stds);
	}
	
	public void resetWells() {
		simPanel.redoWells();
	}
	
	public void clearWells() {
		simPanel.clearWells();
	}
	
	public int getButtonState() {
		return paramPanel.getButtonState();
	}
	
	/**
	 * configure the animation speed
	 *
	 * @param s the animation speed
	 */
	public void setAnimationSpeed(String s) {
		simPanel.setPause(s);
	}

	/**
	 * displayProtein(Protein protein) display protein info on the plotting panel
	 *
	 * @param protein the protein to display
	 */
	public void displayProtein(Protein protein) {
		dataPanel.displayData(protein);
	}

	static public void main(String[] args) {
		JFrame f = new JFrame();
		f.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
		f.add(new Electrophoresis(f));
		f.pack();
		f.setVisible(true);
	}

	

}

File: site/swingjs/j2s/main/java/Electro1D/Electrophoresis.js
(function(){var P$=Clazz.newPackage("main.java.Electro1D"),I$=[[0,'javax.swing.JPanel','javax.swing.JButton','main.java.Utilities.BrowserLauncher','java.awt.Dimension','main.java.Utilities.GenomeFileParser','main.java.Electro1D.Simulation','main.java.Electro1D.Parameters','main.java.Electro1D.ProteinData','main.java.Electro1D.Plot','java.awt.GridBagLayout','java.awt.GridBagConstraints','java.awt.FlowLayout','javax.swing.JTabbedPane','java.awt.Insets','javax.swing.JFrame']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "Electrophoresis", null, 'javax.swing.JPanel');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['O',['tabPane','javax.swing.JTabbedPane','+tabPane2','paramPanel','main.java.Electro1D.Parameters','simPanel','main.java.Electro1D.Simulation','dataPanel','main.java.Electro1D.ProteinData','plotPanel','main.java.Electro1D.Plot','parentFrame','javax.swing.JFrame']]]

Clazz.newMeth(C$, 'Header$',  function () {
var headPanel=Clazz.new_($I$(1,1));
var help=Clazz.new_($I$(2,1).c$$S,["Help"]);
help.setToolTipText$S("Open Help wiki for Electro1D");
help.addActionListener$java_awt_event_ActionListener(((P$.Electrophoresis$1||
(function(){/*a*/var C$=Clazz.newClass(P$, "Electrophoresis$1", function(){Clazz.newInstance(this, arguments[0],1,C$);}, null, 'java.awt.event.ActionListener', 1);

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

Clazz.newMeth(C$, 'actionPerformed$java_awt_event_ActionEvent',  function (e) {
var url="http://sourceforge.net/p/jbf/wiki/Electro1D";
try {
$I$(3).openURL$S(url);
} catch (i) {
if (Clazz.exceptionOf(i,"java.io.IOException")){
System.err.println$S(i.getMessage$());
} else {
throw i;
}
}
});
})()
), Clazz.new_(P$.Electrophoresis$1.$init$,[this, null])));
headPanel.add$java_awt_Component(help);
C$.superclazz.prototype.add$java_awt_Component.apply(this, [headPanel]);
});

Clazz.newMeth(C$, 'c$$javax_swing_JFrame',  function (parentFrame) {
Clazz.super_(C$, this);
C$.superclazz.prototype.setPreferredSize$java_awt_Dimension.apply(this, [Clazz.new_($I$(4,1).c$$I$I,[900, 470])]);
$I$(5).init$();
this.parentFrame=parentFrame;
this.parentFrame.setName$S("Electro1D");
this.simPanel=Clazz.new_($I$(6,1).c$$main_java_Electro1D_Electrophoresis,[this]);
this.paramPanel=Clazz.new_($I$(7,1).c$$main_java_Electro1D_Electrophoresis,[this]);
this.dataPanel=Clazz.new_($I$(8,1).c$$main_java_Electro1D_Electrophoresis,[this]);
this.plotPanel=Clazz.new_($I$(9,1).c$$main_java_Electro1D_Electrophoresis,[this]);
this.setLayout$java_awt_LayoutManager(Clazz.new_($I$(10,1)));
var c=Clazz.new_($I$(11,1));
var tempPanel=Clazz.new_($I$(1,1));
tempPanel.setLayout$java_awt_LayoutManager(Clazz.new_($I$(12,1).c$$I,[0]));
tempPanel.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(4,1).c$$I$I,[240, 450]));
this.paramPanel.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(4,1).c$$I$I,[240, 450]));
this.dataPanel.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(4,1).c$$I$I,[240, 450]));
this.tabPane=Clazz.new_($I$(13,1));
this.tabPane.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(4,1).c$$I$I,[240, 450]));
this.tabPane.addTab$S$java_awt_Component("Parameters", this.paramPanel);
this.tabPane.addTab$S$java_awt_Component("ProteinData", this.dataPanel);
tempPanel.add$java_awt_Component(this.tabPane);
var tempPanel2=Clazz.new_($I$(1,1));
tempPanel2.setLayout$java_awt_LayoutManager(Clazz.new_($I$(12,1).c$$I,[0]));
tempPanel2.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(4,1).c$$I$I,[650, 450]));
this.simPanel.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(4,1).c$$I$I,[650, 450]));
this.plotPanel.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(4,1).c$$I$I,[650, 450]));
this.tabPane2=Clazz.new_($I$(13,1));
this.tabPane2.setPreferredSize$java_awt_Dimension(Clazz.new_($I$(4,1).c$$I$I,[650, 450]));
this.tabPane2.add$S$java_awt_Component("Casting Tray ", this.simPanel);
this.tabPane2.add$S$java_awt_Component("Plot", this.plotPanel);
tempPanel2.add$java_awt_Component(this.tabPane2);
c.gridx=0;
c.gridy=0;
this.add$java_awt_Component$O(this.tabPane, c);
c.gridx=1;
c.gridy=0;
c.insets=Clazz.new_($I$(14,1).c$$I$I$I$I,[0, 5, 0, 0]);
this.add$java_awt_Component$O(this.tabPane2, c);
this.paramPanel.setDefaults$();
}, 1);

Clazz.newMeth(C$, 'addStandard$',  function () {
this.simPanel.addStandard$();
});

Clazz.newMeth(C$, 'stopRun$',  function () {
this.simPanel.stopRun$();
});

Clazz.newMeth(C$, 'displaySim$',  function () {
this.tabPane.setVisible$Z(true);
});

Clazz.newMeth(C$, 'setAcrylamide$main_java_Electro1D_Acrylamide',  function (acrylamide) {
this.simPanel.setAcrylamide$main_java_Electro1D_Acrylamide(acrylamide);
});

Clazz.newMeth(C$, 'startRun$main_java_Electro1D_ProteinA$main_java_Electro1D_Protein$main_java_Electro1D_Protein$main_java_Electro1D_ProteinA',  function (aprotein, protein1, protein2, proteins) {
this.simPanel.startRun$main_java_Electro1D_ProteinA$main_java_Electro1D_Protein$main_java_Electro1D_Protein$main_java_Electro1D_ProteinA(aprotein, protein1, protein2, proteins);
});

Clazz.newMeth(C$, 'setPlotData$main_java_Electro1D_ProteinA$main_java_Electro1D_Protein$main_java_Electro1D_Protein',  function (stds, sample, dye) {
this.tabPane2.setEnabledAt$I$Z(this.tabPane2.indexOfComponent$java_awt_Component(this.plotPanel), this.plotPanel.setResults$main_java_Electro1D_ProteinA$main_java_Electro1D_Protein$main_java_Electro1D_Protein(stds, sample, dye));
});

Clazz.newMeth(C$, 'displayData$',  function () {
this.tabPane2.setVisible$Z(true);
});

Clazz.newMeth(C$, 'addSample$main_java_Electro1D_ProteinA',  function (stds) {
this.simPanel.addSample$main_java_Electro1D_ProteinA(stds);
});

Clazz.newMeth(C$, 'addSample2$',  function () {
this.simPanel.addSample2$();
});

Clazz.newMeth(C$, 'addSampleFromFile$java_util_Vector$I',  function (proteins, wellNum) {
this.simPanel.increaseDDNum$();
this.simPanel.addSampleFromFile$java_util_Vector$I(proteins, wellNum);
});

Clazz.newMeth(C$, 'isReady$',  function () {
return this.simPanel.isReady$();
});

Clazz.newMeth(C$, 'paramsetspeed$D',  function (d) {
this.paramPanel.setSpeed$D(d);
});

Clazz.newMeth(C$, 'updateSpeed$D$main_java_Electro1D_Acrylamide',  function (d, acrgel) {
this.simPanel.updateSpeed$D$main_java_Electro1D_Acrylamide(d, acrgel);
});

Clazz.newMeth(C$, 'setStandards$main_java_Electro1D_ProteinA',  function (stds) {
this.simPanel.setStandards$main_java_Electro1D_ProteinA(stds);
});

Clazz.newMeth(C$, 'resetWells$',  function () {
this.simPanel.redoWells$();
});

Clazz.newMeth(C$, 'clearWells$',  function () {
this.simPanel.clearWells$();
});

Clazz.newMeth(C$, 'getButtonState$',  function () {
return this.paramPanel.getButtonState$();
});

Clazz.newMeth(C$, 'setAnimationSpeed$S',  function (s) {
this.simPanel.setPause$S(s);
});

Clazz.newMeth(C$, 'displayProtein$main_java_Electro1D_Protein',  function (protein) {
this.dataPanel.displayData$main_java_Electro1D_Protein(protein);
});

Clazz.newMeth(C$, 'main$SA',  function (args) {
var f=Clazz.new_($I$(15,1));
f.setDefaultCloseOperation$I(2);
f.add$java_awt_Component(Clazz.new_(C$.c$$javax_swing_JFrame,[f]));
f.pack$();
f.setVisible$Z(true);
}, 1);

Clazz.newMeth(C$);
})();
;Clazz.setTVer('3.3.1-v4');//Created 2022-04-21 16:53:53 Java2ScriptVisitor version 3.3.1-v4 net.sf.j2s.core.jar version 3.3.1-v4
